### ISSUE TITLE
MindMajix is the globally professional in IT courses training which emphasize on hands-on experience with examples from real-time scenarios by experts. It is the largest provider of high quality online training courses.  It is conceptualized and initiated by several multidisciplinary and ingenious software technocrats having a combined experience of more than 10 yrs in the industry.

### ISSUE THREAD

**Issue by emmablisa** [emmablisa/mindmajix#1]
I am developing opensouce project with a huge pile of dependencies, and I find Puppet a perfect tool for my requirements.

However, my target audience is not necessarily experienced with Puppet. To the point they may not be even know how web servers work. So I would like to ask the user to “run this in terminal” for the simplest case, but if the user also wants to configure some parts of my software, he would have to learn puppet. And educating them Puppet defeats the “servers for dummies” purpose of my project.

Basically, dare I say, I need Unity from the deployment world.

The thing I came up with, is to use ‘json’ as a backend for [Puppet training](https://mindmajix.com/puppet-training), and then write some GUI tool that would generate/modify that json using JSON Schema. Of couse, this thing will be optional and experient Puppet usets could still just use my module as is.

The questiond is, am I doing it right? Is there a solution for my problem?
